src/Perf/IIDOptimizer/SignatureGenerator.cs

```diff
@@ -183,7 +182,7 @@ public SignaturePart GetSignatureParts(TypeReference type)
 
         private TypeReference CreateAuthoringMetadataTypeReference(TypeReference type)
         {
-            return new TypeReference($"ABI.Impl.{type.Name}", type.Name, assembly.MainModule, type.Module);
+            return new TypeReference($"ABI.Impl.{type.Name}", type.Name, type.Module, assembly.MainModule);

```

Comment:
Should the first parameter be namespace?

Reply:
I'd say so